pkg/daemon/ceph/osd/daemon.go

```diff
@@ -104,17 +104,16 @@ func StartOSD(context *clusterd.Context, osdType, osdID, osdUUID, lvPath string,
 	return nil
 }
 
-func handleTerminate(context *clusterd.Context, lvPath, volumeGroupName string) error {
+func handleTerminate(context *clusterd.Context, lvPath, volumeGroupName string) {
 	sigc := make(chan os.Signal, 1)
 	signal.Notify(sigc, syscall.SIGTERM)
 
 	<-sigc
 	logger.Infof("shutdown signal received, exiting...")
 	err := killCephOSDProcess(context, lvPath)
 	if err != nil {
-		return errors.Wrap(err, "failed to kill ceph-osd process")
+		logger.Error("failed to kill ceph-osd process. %v", err)
```

Comment:
```suggestion
		logger.Errorf("failed to kill ceph-osd process. %v", err)
```

Since when did we decide not to fail here?

Reply:
Earlier also, we were just returning error but we are using this method `handleTerminate` in a go routine.